src/queue.js

```diff
@@ -0,0 +1,44 @@
+export default class Queue {
+  constructor() {
+    this.elements = []
+    this.size = 0
+  }
+
+  enqueue(element) {
+    return this.elements[this.size++] = element
+  }
+
+  dequeue() {
+    if (this.size === 0) {
+      return null
+    } else {
+      const dequeuedElement = this.elements[0]
+      this.elements.splice(0,1)
```

Comment:
Splice returns the removed elements, so these two lines could be combined into:

const dequeuedElement = this.elements.splice(0, 1)